### PR TITLE
Fabric Video: position label and constrain video

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Positions the label along the grid when a fabric advert is delivered in the top slot.

fluid-video

Although this is not shown here, the advert has a fixed 250 height and the video will cover this height and get clipped if the window gets resized horizontally, thanks to the `object-fit` property.

cc @guardian/commercial-dev

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/dfp/private/render-advert.js [hunk 1]
```diff
@@ -31,12 +31,28 @@ define([
      *
      */
 
+    function addClassIfHasClass(newClassNames) {
```
Comment (JonNorman) [inline static/src/javascripts/projects/common/modules/commercial/dfp/private/render-advert.js]: would it be worth naming each of these functions to make it clearer what each is doing?

File: static/src/stylesheets/module/_adslot.scss [hunk 2]
```diff
@@ -135,12 +135,46 @@
         padding-top: $gs-row-height / 2;
 
         /* Oh horrible code, I'm looking at you. One day, you'll disappear.
-           One day, I'll fucking rip you apart.
+           One day, I'll [redacted].
```
Comment (JonNorman) [inline static/src/stylesheets/module/_adslot.scss]: 😂

File: static/src/javascripts/projects/common/modules/commercial/dfp/private/render-advert.js (hunk 1 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/dfp/private/render-advert.js]: sure!

**Comment by JonNorman** [comment]
Can't speak too strongly for the CSS but everything else looks good to me :+1:

(might want to look at that failing test though...)

**Pull request merged**